File: GYM.Controller.Api/DTOs/TrainingDTO.cs
using GYM.Data.Entities;
namespace GYM.Controller.Api.DTOs;

public class TrainingDTO
{
    public int? Id {get;set;}
    public string TrainingName {get;set;} = default!;
    public string Difficulty{get;set;} = default!;
    public Place Place {get;set;}
    public int Calories {get;set;}
    public string Description {get;set;} = default!;
    public TimeOnly EstimatedTime{get;set;}
    public DateTime? CreatedAt{get;set;}
    public List<ExerciseDTO> Exercises {get;init;} = default!;

}

File: GYM.FrontEnd/src/types/trainingDTO.ts
import type { exerciseDTO } from "./ExerciseDTO";

export type Place = 'Home' | 'Gym' | 'Outdoor';

export interface TrainingDTO {
  id?: number;                  // int? -> opcional (number)
  trainingName: string;        // string -> string
  difficulty: string;          // string -> string
  place: Place | number;      // Place -> Enum o número
  calories: number;            // int -> number
  description: string;         // string -> string
  estimatedTime: string;       // TimeOnly -> string (ej. "01:30:00")
  createdAt?: string;          // DateTime? -> opcional string en formato ISO (ej. "2026-07-22T...")
  exercises: exerciseDTO[];    // List<ExerciseDTO> -> Array de ExerciseDTO
}

export interface TrainingCreateDTO {
    id?: number;
    trainingName: string;        // string -> string
    difficulty: string;          // string -> string
    place: Place | number;      // Place -> Enum o número
    calories: number;            // int -> number
    description: string;         // string -> string
    estimatedTime: string;       // TimeOnly -> string (ej. "01:30:00")
    exercisesIDs: number[];      // List<int> -> Array de números (IDs de ejercicios)
}

